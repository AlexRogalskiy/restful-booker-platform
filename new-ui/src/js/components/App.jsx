import React from "react";
import { Switch, Route, Link } from 'react-router-dom'
import Hotel from './Hotel.jsx';
import Search from './Search.jsx';

export default class App extends React.Component {

    constructor() {
        super();
    }

    render() {
      return(
        <div>
          <h1>Setup</h1>
          <Link to="/hotel">Hotel</Link>
          <Link to="/search">Search</Link>
          <Switch>
            <Route path='/hotel' component={Hotel}/>
            <Route path='/search' component={Search}/>
          </Switch>
        </div>
      );
    }
  }