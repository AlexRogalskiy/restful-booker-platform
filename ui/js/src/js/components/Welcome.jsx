import React from 'react';
import Popup from "reactjs-popup";
import Cookies from 'universal-cookie';
import glassIcon from '../../images/glass.png';
import bracketsIcon from '../../images/brackets.png';
import blockIcon from '../../images/block.png';
import goArrowIcon from '../../images/goarrow.png';

export default class Welcome extends React.Component {

    constructor(){
        super();

        this.state = {
            page : 0
        }

        this.closeModal = this.closeModal.bind(this);
        this.increasePage = this.increasePage.bind(this);
        this.decreasePage = this.decreasePage.bind(this);
    }

    increasePage(){
        const up = this.state.page + 1;

        this.setState({
            page : up
        })
    }

    decreasePage(){
        const down = this.state.page - 1;

        this.setState({
            page : down
        })
    }

    closeModal(){
        const cookies = new Cookies();
        cookies.set('welcome', true, { path: '/' });

        this.props.setWelcome(false);
    }

    render(){
		let content = null;
        let buttons = null;

        switch(this.state.page){
            case 0:
                content = <div className="col-sm-10 content">
                            <br />
                            <br />
                            <h2>Welcome to Shady Meadows Booking Manager</h2>
                            <br />
                            <br />
                            <h4>Your one stop shop to practise Software Testing!</h4>
                          </div>

                buttons = <div>
                            <div className="col-sm-4"></div>
                            <div className="col-sm-2 text-center">
                                <button id="next" className="btn btn-primary" onClick={this.increasePage}>Next</button>
                            </div>
                          </div>
                break;
            case 1: 
                content = <div className="col-sm-10 content">
                            <br />
                            <img style={{height : "60px", marginLeft : "auto", marginRight : "auto", display : "block"}} src={glassIcon} alt="Magnifying glass" />
                            <br />
                            <p><span style={{fontWeight: "bold"}}>Exploration: </span> Testing is more than just finding bugs. With Shady Meadows Booking Manager you take time to hone your exploratory testing skills by diving into the application to find out more about how the application works. There are many features for you to explore, with more being continuously added for future. So there is always more to explore!</p>
                          </div>

                buttons = <div>
                            <div className="col-sm-2 text-center">
                                <button id="prev" className="btn btn-primary" onClick={this.decreasePage}>Prev</button>
                            </div>
                            <div className="col-sm-2"></div>
                            <div className="col-sm-2 text-center">
                                <button id="next" className="btn btn-primary" onClick={this.increasePage}>Next</button>
                            </div>
                          </div>
                break;
            case 2:
                content = <div className="col-sm-10 content">
                            <br />
                            <img style={{height : "60px", marginLeft : "auto", marginRight : "auto", display : "block"}} src={bracketsIcon} alt="Brackets" />
                            <br />
                            <p><span style={{fontWeight: "bold"}}>Automation:</span> Shady Meadows Booking Manager is an instance of the Open source application, restful-booker-platform. RBP offers a range of different technologies that you automate against, either online or via a locally deployed instance.</p>
                            <p>Check out the <a href="https://github.com/mwinteringham/restful-booker-platform" alt="Restful booker platform source code">restful-booker-platform source code</a> to learn more about the various APIs and JavaScript features to practise your Automation in Testing skills.</p>
                          </div>

                buttons = <div>
                            <div className="col-sm-2 text-center">
                                <button id="prev" className="btn btn-primary" onClick={this.decreasePage}>Prev</button>
                            </div>
                            <div className="col-sm-2"></div>
                            <div className="col-sm-2 text-center">
                                <button id="next" className="btn btn-primary" onClick={this.increasePage}>Next</button>
                            </div>
                          </div>
                break;
            case 3:
                content = <div className="col-sm-10 content">
                            <br />
                            <img style={{height : "60px", marginLeft : "auto", marginRight : "auto", display : "block"}} src={blockIcon} alt="Building blocks" />
                            <br />
                            <p><span style={{fontWeight: "bold"}}>Infrastructure:</span> Shady Meadows Booking Manager is a continuously deployed application using CircleCi, Docker and Kubernetes. All the deployment assets can be found in the <a href="https://github.com/mwinteringham/restful-booker-platform" alt="Restful booker platform source code">restful-booker-platform source</a> repository for you to create your own pipeline. You can also learn more about the <a href="https://circleci.com/gh/mwinteringham/workflows/restful-booker-platform" alt="Circle CI build pipeline">build process in this public build pipeline</a>.</p>
                        </div>

                buttons = <div>
                            <div className="col-sm-2 text-center">
                                <button id="prev" className="btn btn-primary" onClick={this.decreasePage}>Prev</button>
                            </div>
                            <div className="col-sm-2"></div>
                            <div className="col-sm-2 text-center">
                                <button id="next" className="btn btn-primary" onClick={this.increasePage}>Next</button>
                            </div>
                          </div>                
                break;
            case 4:
                content = <div className="col-sm-10 content">
                            <br />
                            <img style={{height : "60px", marginLeft : "auto", marginRight : "auto", display : "block"}} src={goArrowIcon} alt="Go arrow" />
                            <br />
                            <p><span style={{fontWeight: "bold"}}>Get Started:</span> How you use this application is up to you, but here are a few things to get you started:</p>
                            <ul>
                                <li>To log into the application use the credentials admin/password</li>
                                <li>You can <a href="https://github.com/mwinteringham/restful-booker-platform/projects/1" alt="Link to RBP project">read more about the features here</a></li>
                                <li>If you find a particularly bad bug, <a href="https://github.com/mwinteringham/restful-booker-platform/issues" alt="Bug tracker">feel free to raise it here</a></li>
                            </ul>
                            <p><span style={{fontWeight: "bold"}}>Please note:</span> for security purposes the database resets itself every 10 minutes.</p>
                          </div>

                buttons = <div>
                            <div className="col-sm-2 text-center">
                                <button id="prev"  className="btn btn-primary" onClick={this.decreasePage}>Prev</button>
                            </div>
                            <div className="col-sm-2"></div>
                            <div className="col-sm-2 text-center">
                                <button id="closeModal" className="btn btn-success" onClick={this.closeModal}>Close</button>
                            </div>
                          </div>
                break;
        }

        return(
            <Popup defaultOpen={true} closeOnEscape={false} modal closeOnDocumentClick={false}>
                <div className="modal-body welcome">
                    <div className="container-fluid">
                        <div className="row">
                            <div className="col-sm-1"></div>
                            {content}
                            <div className="col-sm-1"></div>
                        </div>
                        <div className="row justify-content-center">
                            <div className="col-sm-3"></div>
                                {buttons}
                            <div className="col-sm-3"></div>
                        </div>
                    </div>
                </div>
            </Popup>
        )
    }

}