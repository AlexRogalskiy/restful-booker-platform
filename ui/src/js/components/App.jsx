import React from 'react';
import { Route, Switch } from 'react-router-dom'
import RoomListings from './RoomListings.jsx';
import Search from './Search.jsx';
import Nav from './Nav.jsx';
import Login from './Login.jsx'
import RoomDetails from './RoomDetails.jsx';
import Report from './Report.jsx';
import Cookies from 'universal-cookie';
import { API_ROOT } from '../api-config';

export default class App extends React.Component {

    constructor() {
        super();

        this.state = {
            isAuthenticated : false
        }
        
        this.setAuthenticate = this.setAuthenticate.bind(this);
    }

    componentDidMount(){
        const cookies = new Cookies();

        fetch(API_ROOT.auth + '/validate', {
                method: 'POST',
                headers: {
                    'Accept': 'application/json',
                    'Content-Type': 'application/json'
                },
                body : JSON.stringify({ token: cookies.get('token')})
            })
            .then(res => {
                if(res.status == 200){
                    this.setState({isAuthenticated: true});
                }
            })
    }

    setAuthenticate(e){
        this.setState({
            isAuthenticated : e
        });
    }

    render() {
        let app = null;

        if(!this.state.isAuthenticated){
            app = <Login setAuthenticate={this.setAuthenticate}/>
        } else {
            app = <div>
                    <Nav setAuthenticate={this.setAuthenticate} />
                    <Switch>
                        <Route exact path='/' render={(props) => <RoomListings {...props} />} />
                        <Route exact path='/search' component={Search} {...this.props}/>
                        <Route exact path='/room/:id' render={({ location, match }) => (
                            <RoomDetails isAuthenticated={this.state.isAuthenticated} params={match.params}/>
                        )} />
                        <Route exact path='/report' component={Report} />
                    </Switch>
                </div>
        }

        return(
            <div className="container">
                {app}
            </div>
        );
    }
}
