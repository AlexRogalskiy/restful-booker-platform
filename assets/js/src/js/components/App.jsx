import React from 'react';
import { HashRouter, Route, Switch } from 'react-router-dom'
import { SHOW_WELCOME } from '../api-config';
import ReactModal from 'react-modal';
import Cookies from 'universal-cookie';
import { createBrowserHistory as createHistory } from 'history'
import ReactGA from 'react-ga';

import AdminContainer from './AdminContainer.jsx';
import Welcome from './Welcome.jsx';
import CookiePolicy from './CookiePolicy.jsx';
import PrivacyPolicy from './PrivacyPolicy.jsx';
import Home from './Home.jsx';
import Footer from './Footer.jsx';

if (process.env.NODE_ENV !== 'test') ReactModal.setAppElement('#root');

const history = createHistory();

ReactGA.initialize('UA-118712228-3');

history.listen((location) => {
    ReactGA.pageview(location.pathname + location.hash);
});

export default class App extends React.Component {

    constructor() {
        super();

        this.state = {
            showWelcome : false
        }
        
        this.setWelcome = this.setWelcome.bind(this);

        ReactGA.pageview(window.location.pathname + window.location.hash);
    }

    componentDidMount(){
        const cookies = new Cookies();

        if(typeof cookies.get('welcome') === 'undefined' && SHOW_WELCOME){
           this.setState({
                showWelcome : true
            })
        }
    }

    setWelcome(e){
        this.setState({
            showWelcome : e
        });
    }

    render() {
        let welcome = null;

        if(this.state.showWelcome){
            welcome = <div>
                <Welcome setWelcome={this.setWelcome} />
            </div>
        }

        return(
            <HashRouter>
                {welcome}
                    <Switch>
                        <Route path='/admin/' render={() => (
                            <AdminContainer />
                        )} />
                        <Route exact path='/' component={Home} />
                        <Route exact path='/cookie' component={CookiePolicy} />
                        <Route exact path='/privacy' component={PrivacyPolicy} />
                    </Switch>
                <Footer />
            </HashRouter>
        );
    }
}
