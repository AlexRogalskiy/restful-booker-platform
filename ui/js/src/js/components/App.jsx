import React, { Suspense } from 'react';
import { Route, Switch } from 'react-router-dom'
import { API_ROOT, SHOW_WELCOME } from '../api-config';

import Nav from './Nav.jsx';
import Login from './Login.jsx'
import Report from './Report.jsx';
import Welcome from './Welcome.jsx';
import Cookies from 'universal-cookie';
import CookiePolicy from './CookiePolicy.jsx';
import PrivacyPolicy from './PrivacyPolicy.jsx';
import Loading from './Loading.jsx';
import Home from './Home.jsx';
import Branding from './Branding.jsx';
import Footer from './Footer.jsx';
import ReactModal from 'react-modal';

const RoomListings = React.lazy(() => import('./RoomListings.jsx'));
const RoomDetails = React.lazy(() => import('./RoomDetails.jsx'));

ReactModal.setAppElement('#root');

export default class App extends React.Component {

    constructor() {
        super();

        this.state = {
            isAuthenticated : false,
            showWelcome : false
        }
        
        this.setAuthenticate = this.setAuthenticate.bind(this);
        this.setWelcome = this.setWelcome.bind(this);
    }

    componentDidMount(){
        const cookies = new Cookies();

        if(typeof cookies.get('welcome') === 'undefined' && SHOW_WELCOME){
           this.setState({
                isAuthenticated : this.state.isAuthenticated,
                showWelcome : true
            })
        }

        fetch(API_ROOT.auth + '/validate', {
                method: 'POST',
                headers: {
                    'Accept': 'application/json',
                    'Content-Type': 'application/json'
                },
                body : JSON.stringify({ token: cookies.get('token')})
            })
            .then(res => {
                if(res.status == 200){
                    this.setAuthenticate(true);
                }
            })
    }

    setAuthenticate(e){
        this.setState({
            isAuthenticated : e,
            showWelcome : this.state.showWelcome
        });
    }

    setWelcome(e){
        this.setState({
            isAuthenticated : this.state.isAuthenticated,
            showWelcome : e
        });
    }

    render() {
        let welcome = null;

        if(this.state.showWelcome){
            welcome = <div>
                <Welcome setWelcome={this.setWelcome} />
            </div>
        }

        return(
            <div>
                {welcome}
                    <Switch>
                        <Route path='/admin/' render={() => (
                            <div>
                                <Nav setAuthenticate={this.setAuthenticate} isAuthenticated={this.state.isAuthenticated} />
                                <div className="container">
                                    <div>
                                        {this.state.isAuthenticated ? (
                                            <div>
                                                <Route exact path='/admin/' render={(props) => (
                                                    <div>
                                                        <Suspense fallback={<Loading />}>
                                                            <RoomListings {...props} />
                                                        </Suspense>
                                                    </div>
                                                )} />
                                                <Route exact path='/admin/room/:id' render={({ location, match }) => (
                                                    <div>
                                                        <Suspense fallback={<Loading />}>
                                                            <RoomDetails params={match.params}/>
                                                        </Suspense>
                                                    </div>
                                                )} />
                                                <Route exact path='/admin/report' component={Report} />
                                                <Route exact path='/admin/branding' component={Branding} />
                                            </div>
                                        ) : (
                                            <div>
                                                <Login setAuthenticate={this.setAuthenticate} />
                                            </div>
                                        )}
                                    </div>
                                </div>
                            </div>
                        )} />
                        <Route exact path='/' component={Home} />
                        <Route exact path='/cookie' component={CookiePolicy} />
                        <Route exact path='/privacy' component={PrivacyPolicy} />
                    </Switch>
                <Footer />
            </div>
        );
    }
}
