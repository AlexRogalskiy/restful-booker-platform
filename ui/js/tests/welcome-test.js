import React from 'react';
import Welcome from '../src/js/components/Welcome.jsx';
import App from '../src/js/components/App.jsx';
import nock from 'nock';

beforeEach(() => {
    nock('http://localhost:3004')
        .post('/auth/validate')
        .reply(200);
});

test('Welcome component contains correct details', () => {
    const welcomeComponent = shallow(
        <Welcome />
    )

    expect(welcomeComponent).toMatchSnapshot();
});

test('Welcome component appears on initial visit', () => {
    const appComponent = shallow(
        <App />
    )

    expect(appComponent).toMatchSnapshot();
});

test('Welcome component doesnt appear after close', () => {
    const appComponent = shallow(
        <App />
    )

    appComponent.instance().setWelcome(false);

    expect(appComponent).toMatchSnapshot();
});

test('Clicking on close button closes modal', () => {
    const appComponent = mount(
        <App />
    )

    appComponent.find('Welcome').instance().setState({
        page : 4
    });

    appComponent.update();

    appComponent.find('#closeModal').simulate('click');

    expect(appComponent).toMatchSnapshot();
});