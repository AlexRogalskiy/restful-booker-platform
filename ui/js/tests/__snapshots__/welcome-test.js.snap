// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking on close button closes modal 1`] = `
<App>
  <div
    className="container"
  >
    <div>
      <Login
        setAuthenticate={[Function]}
      >
        <div
          style={
            Object {
              "marginTop": "20%",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-sm-2"
            />
            <div
              className="col-sm-8"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <h2>
                Welcome to the Shady Meadows 
                <br />
                 booking management system
              </h2>
            </div>
            <div
              className="col-sm-2"
            />
          </div>
          <form>
            <div
              className="row"
            >
              <div
                className="col-sm-4"
              />
              <div
                className="col-sm-4"
              >
                <p>
                  <label
                    htmlFor="username"
                  >
                    Username 
                  </label>
                  <input
                    id="username"
                    onChange={[Function]}
                    style={
                      Object {
                        "border": "1px solid grey",
                      }
                    }
                    type="text"
                  />
                </p>
              </div>
              <div
                className="col-sm-4"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col-sm-4"
              />
              <div
                className="col-sm-4"
              >
                <p>
                  <label
                    htmlFor="password"
                  >
                    Password 
                  </label>
                  <input
                    id="password"
                    onChange={[Function]}
                    style={
                      Object {
                        "border": "1px solid grey",
                      }
                    }
                    type="password"
                  />
                </p>
              </div>
              <div
                className="col-sm-4"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col-sm-7"
              />
              <div
                className="col-sm-1"
              >
                <button
                  className="btn btn-default"
                  id="doLogin"
                  onClick={[Function]}
                  style={
                    Object {
                      "marginLeft": "9px",
                    }
                  }
                  type="button"
                >
                  Login
                </button>
              </div>
              <div
                className="col-sm-4"
              />
            </div>
          </form>
        </div>
      </Login>
    </div>
  </div>
</App>
`;

exports[`Welcome component appears on initial visit 1`] = `
<div
  className="container"
>
  <div>
    <Welcome
      setWelcome={[Function]}
    />
  </div>
  <div>
    <Login
      setAuthenticate={[Function]}
    />
  </div>
</div>
`;

exports[`Welcome component contains correct details 1`] = `
<Popup
  arrow={true}
  arrowStyle={Object {}}
  className=""
  closeOnDocumentClick={false}
  closeOnEscape={false}
  contentStyle={Object {}}
  defaultOpen={true}
  keepTooltipInside={false}
  lockScroll={false}
  modal={true}
  mouseEnterDelay={100}
  mouseLeaveDelay={100}
  offsetX={0}
  offsetY={0}
  on={
    Array [
      "click",
    ]
  }
  onClose={[Function]}
  onOpen={[Function]}
  open={false}
  overlayStyle={Object {}}
  position="bottom center"
  trigger={null}
>
  <div
    className="modal-body welcome"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-1"
        />
        <div
          className="col-sm-10 content"
        >
          <br />
          <br />
          <h2>
            Welcome to Shady Meadows Booking Manager
          </h2>
          <br />
          <br />
          <h4>
            Your one stop shop to practise Software Testing!
          </h4>
        </div>
        <div
          className="col-sm-1"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        />
        <div>
          <div
            className="col-sm-4"
          />
          <div
            className="col-sm-2"
          >
            <button
              className="btn btn-primary"
              id="next"
              onClick={[Function]}
            >
              Next
            </button>
          </div>
        </div>
        <div
          className="col-sm-3"
        />
      </div>
    </div>
  </div>
</Popup>
`;

exports[`Welcome component doesnt appear after close 1`] = `
<div
  className="container"
>
  <div>
    <Login
      setAuthenticate={[Function]}
    />
  </div>
</div>
`;
